app/src/main/java/org/simple/clinic/facilitypicker/FacilityPickerUiRenderer.kt

```diff
@@ -1,11 +1,53 @@
 package org.simple.clinic.facilitypicker
 
+import com.squareup.inject.assisted.Assisted
+import com.squareup.inject.assisted.AssistedInject
+import org.simple.clinic.facility.change.FacilityListItemBuilder
+import org.simple.clinic.location.LocationUpdate
 import org.simple.clinic.mobius.ViewRenderer
+import org.simple.clinic.registration.RegistrationConfig
 
-class FacilityPickerUiRenderer(
-    private val ui: FacilityPickerUi
+class FacilityPickerUiRenderer @AssistedInject constructor(
+    private val facilityListItemBuilder: FacilityListItemBuilder,
+    private val registrationConfig: RegistrationConfig,
+    @Assisted private val ui: FacilityPickerUi
 ) : ViewRenderer<FacilityPickerModel> {
 
+  @AssistedInject.Factory
+  interface Factory {
+    fun create(ui: FacilityPickerUi): FacilityPickerUiRenderer
+  }
+
   override fun render(model: FacilityPickerModel) {
+    if (model.hasLoadedFacilities) {
+      ui.hideProgressIndicator()
+      renderFacilities(model)
+    } else {
+      ui.showProgressIndicator()
+    }
+
+    if (model.hasLoadedTotalFacilityCount) {
+      if (model.totalFacilityCount!! > 0) {
```

Comment:
Couldn't we have checked for facilities list size instead? 🤔. That way we could avoid doing another DB query to get the count. Are we doing this to be more explicit?

Reply:
Yes, they are two different things. In addition, we do not want this functionality to break if we move the screen to paging library later.